reascripts/ReaSpeech/source/TranscriptExporter.lua

```diff
@@ -45,6 +52,22 @@ function TranscriptExporter:init()
   self.alert_popup = AlertPopup.new {}
 end
 
+function TranscriptExporter:open()
+  local transparent, opaque, duration = 0.0, 1.0, 0.2
+
+  local on_done = function()
+    -- set the style to a static value
+    self.theme.styles[1][2] = opaque
```

Comment:
Maybe we could use a reference to avoid the assumption that the window opacity is the first style. Alternatively, we could use a map with keys like `ImGui.StyleVar_Alpha` in place of a list.

Reply:
had a similar feeling. for the purposes of this example i thought it was helpful to be explicit about how it's patching an `ImGuiTheme` on start and resetting it to a static value on completion.

related to the other comment (~ applying it to other popups), i'm thinking through what (if any) implications there are about reusing a theme definition. even just typing it out there i'm feeling like if there's a central popup-related theme that any instances using it would want to create their own copy (differing from the global `Theme` instance that's applied to the main window and inherited by any child windows). 🤔

Reply:
I think there should be one main theme, and a derived theme for popups. Being able to adjust the theme(s) in one place is what makes them useful in my opinion. Having a central definition of the theme(s) helps ensure consistency across the application and minimizes the amount of redundant changes that need to be made. It's fine to have the flexibility of a different theme per window, but in practice, I think it's best to minimize use of that flexibility.